mvc-login/src/main/java/com/auth0/example/SecurityConfig.java

```diff
@@ -1,41 +1,45 @@
 package com.auth0.example;
 
+import org.springframework.beans.factory.annotation.Value;
 import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.config.annotation.method.configuration.EnableMethodSecurity;
 import org.springframework.security.config.annotation.web.builders.HttpSecurity;
-import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
 import org.springframework.security.web.SecurityFilterChain;
-import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+import org.springframework.security.web.authentication.logout.LogoutHandler;
+import org.springframework.web.servlet.support.ServletUriComponentsBuilder;
 
-@EnableWebSecurity
-public class SecurityConfig {
+import java.io.IOException;
 
-    private final LogoutHandler logoutHandler;
+import static org.springframework.security.config.Customizer.withDefaults;
 
-    public SecurityConfig(LogoutHandler logoutHandler) {
-        this.logoutHandler = logoutHandler;
-    }
+@Configuration
+@EnableMethodSecurity(securedEnabled = true)
+public class SecurityConfig {
+
+    @Value("${okta.oauth2.issuer}")
+    private String issuer;
+    @Value("${okta.oauth2.client-id}")
+    private String clientId;
 
     @Bean
-    public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
-        http.authorizeRequests()
-                // allow all users to access the home pages and the static images directory
-                .mvcMatchers("/", "/images/**").permitAll()
-                // all other requests must be authenticated
-                .anyRequest().authenticated()
-                .and().oauth2Login()
-                .and().logout()
-                // handle logout requests at /logout path
-                .logoutRequestMatcher(new AntPathRequestMatcher("/logout"))
-                // customize logout handler to log out of Auth0
-                .addLogoutHandler(logoutHandler);
-        return http.build();
+    public SecurityFilterChain configure(HttpSecurity http) throws Exception {
+        return http.authorizeHttpRequests((requests) -> requests
+                .requestMatchers("/", "/images/**").permitAll()
+                    .anyRequest().authenticated())
+                .oauth2Login(withDefaults())
+                .logout(logout -> logout.addLogoutHandler(logoutHandler()))
+                .build();
```

Comment:
I'm not sure the indents in this code block line up as expected. From one of our Spring Boot labs:

```java
@Bean
public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
    http
        .authorizeHttpRequests(authorize -> authorize
            .anyRequest().authenticated()
        )
        .oauth2Login(withDefaults())
        .logout(logout -> logout.logoutRequestMatcher(new AntPathRequestMatcher("/logout"))
            .addLogoutHandler(oidcLogoutHandler()));
    return http.build();
}
```

Reply:
👍 indentation fixed.